src/api/signing.ts

```diff
@@ -0,0 +1,98 @@
+import { Constants } from '..';
+import {
+  DEFAULT_ETH_DERIVATION,
+  BTC_LEGACY_DERIVATION,
+  BTC_SEGWIT_DERIVATION,
+  BTC_WRAPPED_SEGWIT_DERIVATION,
+  SOLANA_DERIVATION,
+} from '../constants';
+import { queue } from './utilities';
+
+export const sign = async (
+  payload: Uint8Array | Buffer | Buffer[],
+  overrides?: SignRequestParams,
+): Promise<SignData> => {
+  const tx: SignRequestParams = {
+    data: {
+      signerPath: DEFAULT_ETH_DERIVATION,
+      curveType: Constants.SIGNING.CURVES.SECP256K1,
+      hashType: Constants.SIGNING.HASHES.KECCAK256,
+      encodingType: Constants.SIGNING.ENCODINGS.EVM,
+      payload,
+    },
+    ...overrides,
+  };
+  return queue((client) => client.sign(tx));
+};
+
+export const signMessage = async (
+  payload: string,
```

Comment:
I left a comment in the testing section, but yeah I did just check and EIP712 inputs are JS objects, not strings. See `buildEIP712Request`, `buildPersonalSignRequest`, and `buildEthereumMsgRequest`.

Also none of the Ethereum message requests use general signing, which this function appears to use? I'm actually not quite sure what the purpose of this function is... but anyway we do need a way to request ETH messages.